src/util/Views.h

```diff
@@ -138,8 +138,19 @@ auto integerRange(Int upperBound) {
   return std::views::iota(Int{0}, upperBound);
 }
 
-// TODO<joka921> Comments, tests, concepts.
-auto repeatedTransformView(auto view, auto transformation) {
+// TODO<joka921> tests
+// Similar to `std::views::transform` but for transformation functions that
+// transform a value in place. The result always only is an input range,
+// independent of the actual range category of the input.
+template <std::ranges::view View,
+          ad_utility::InvocableWithExactReturnType<
+              void, std::ranges::range_reference_t<View>>
+              Transformation>
+auto inPlaceTransformView(View view, Transformation transformation) {
```

Comment:
I understand how the function works on element pointers instead of directly on the elements and how the same pointer will not be transformed twice. But I don't understand how it can happen that the same pointer is encountered twice in this process,

Reply:
I have added unit tests for exactly this purpose:)